kubetest2/internal/deployers/eksapi/nodegroup.go

```diff
@@ -249,6 +249,56 @@ func (m *NodegroupManager) createUnmanagedNodegroupWithEFA(infra *Infrastructure
 	if err != nil {
 		return err
 	}
+	subnetId := infra.subnetsPrivate[0]
+	var capacityReservationId string
+	if opts.CapacityReservation {
+		capacityReservations, err := m.clients.EC2().DescribeCapacityReservations(context.TODO(), &ec2.DescribeCapacityReservationsInput{
+			Filters: []ec2types.Filter{
+				{
+					Name:   aws.String("instance-type"),
+					Values: opts.InstanceTypes,
+				},
+				{
+					Name:   aws.String("state"),
+					Values: []string{"active"},
+				},
+			},
+		})
+		if err != nil {
+			return fmt.Errorf("failed to describe capacity reservation")
+		}
+		var az string
+		for _, cr := range capacityReservations.CapacityReservations {
+			if *cr.AvailableInstanceCount >= int32(opts.Nodes) {
+				capacityReservationId = *cr.CapacityReservationId
```

Comment:
Do we have capacity reserved cross multiple AZs? If yes, we need to extend this to a list as Carter mentioned, the EFA cross AZ communication is supported.

Reply:
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa.html#efa-limits
> EFA OS-bypass traffic can't cross Availability Zones, VPCs, or AWS accounts.

> We are excited to announce that AWS now supports cross-subnet communication between [Elastic Fabric Adapter (EFA)](https://aws.amazon.com/hpc/efa/) interfaces for Amazon EC2 instances within the same Availability Zone (AZ).

I thought EFA doesn't support cross AZ communication, and I saw the NCCL test fails on it.